tests/baselines/reference/thisTypeInFunctionsNegative.js

```diff
@@ -174,9 +176,13 @@ function initializer(this: C = new C()): number { return this.n; }
 
 // can't name parameters 'this' in a lambda.
 c.explicitProperty = (this, m) => m + this.n;
+const f2 = <T>(this: {n: number}, m: number) => m + this.n;
+const f3 = async (this: {n: number}, m: number) => m + this.n;
+const f4 = async <T>(this: {n: number}, m: number) => m + this.n;
 
 
 //// [thisTypeInFunctionsNegative.js]
+// @target es6

```

Comment:
This seems to have no effect. Otherwise the `__extends` helper wouldn't be emitted.

Reply:
Good catch.  I assumed this was working because I didn't get an error about Promise not being defined, but there may be a different reason that error doesn't occur in the context of the test suite.  It looks like I messed up the syntax.  Fixed.